### PR TITLE
Add feature to allow Aura collator to use full PoV size

### PR DESCRIPTION
**Pull request by s0me0ne-unkn0wn**
This PR introduces a feature that allows to optionally enable using the full PoV size.

Technically, we're ready to enable it by default, but as corresponding runtime changes have not been propagated to the system parachain runtimes yet, doing so could put them at risk. On the other hand, there are teams that could benefit from it right now, and it makes no sense for them to wait for the fellowship release and everything.

### CODE REVIEW

**Comment by xlc** [comment]
why it is a feature flag instead of some parameter passed in?

**Comment by s0me0ne-unkn0wn** [comment]
> why it is a feature flag instead of some parameter passed in?

I just don't see much sense in punching holes in the Aura parameter structures (three of them, to be precise) for something that is obviously transient. The feature allows for keeping the change footprint small.

Besides that, changing the collator parameters is a breaking change. This one with the feature could be backported if need be, but with the parameter, it's not possible.

**Comment by bkchr** [comment]
This should be removed entirely when the time has come. I don't see any reason that we should backport or bring this up as a feature flag. With the next stable release in 5 months we hopefully have ironed out all the bugs we can remove it.

**Comment by bkchr** [comment]
> This should be removed entirely when the time has come. I don't see any reason that we should backport or bring this up as a feature flag. With the next stable release in 5 months we hopefully have ironed out all the bugs we can remove it.

My brain was not working. As benchmarking ensures that we don't run over the PoV limit, I think we can just remove the training wheels. The `PoVReclaim` extension only exists to increase the throughput, but is not a requirement that ensures that without the node side check we don't run into issues. 

So, I would advocate to remove the division by two completely. Nevertheless, we should ensure to backport all the fixes for pov reclaim to ensure that if people are using it, it is secure.

**Review by eskimor — APPROVED** [review]

File: prdoc/pr_5393.prdoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: Allow to enable full PoV size
+
+doc:
+  - audience: Node Dev
+    description: |
+      A feature is introduced allowing a collator to enable full PoV size in compile time.
```
Comment (sandreim) [inline prdoc/pr_5393.prdoc:9]:
```suggestion
      A feature is introduced allowing a collator to enable full PoV size at compile time.
```

**Review by sandreim — APPROVED** [review]

**Pull request merged**